build/gen_stub.php

```diff
@@ -2631,6 +2721,28 @@ function replaceMethodSynopses(string $targetDirectory, array $funcMap, array $a
     return $methodSynopses;
 }
 
+function areXmlsEqual(DOMDocument $doc, DOMElement $originalSynopsis, DOMElement $newSynopsis): bool
+{
+    $docComparator = new DOMDocument();
+    $docComparator->preserveWhiteSpace = false;
+    $docComparator->formatOutput = true;
+
+    $xml1 = $doc->saveXML($originalSynopsis);
+    $xml1 = getReplacedSynopsisXml($xml1);
+    $docComparator->loadXML($xml1);
+    $xml1 = $docComparator->saveXML();
+
+    $originalSynopsis->parentNode->replaceChild($newSynopsis, $originalSynopsis);
```

Comment:
Should this be in here? This looks unrelated to the comparison.

Reply:
Yes, unfortunately it makes the function have a side-effect, but it's still absolutely required (I tried to clone `$originalSynopsis` before in order to prevent this, but it didn't work) :/

The reason why we need this code is that it attaches the `$newSynopsis` element to `$doc` by replacing `$originalSynopsis`.

Reply:
Hm, would using DOMDocument::importNode work? Maybe it's possible to directly import it to $docComparator even?

Reply:
I've just tried this:

```
$docComparator = new DOMDocument();
$docComparator->preserveWhiteSpace = false;
$docComparator->formatOutput = true;
$docComparator->loadXML('<root></root>');

$originalSynopsis = $docComparator->importNode($originalSynopsis, true);
$docComparator->documentElement->appendChild($originalSynopsis);
$xml1 = $docComparator->saveXML($originalSynopsis);

$newSynopsis = $docComparator->importNode($newSynopsis, true);
$docComparator->documentElement->appendChild($newSynopsis);
$newSynopsis->setAttribute("xmlns", "http://docbook.org/ns/docbook");
$xml2 = $docComparator->saveXML($newSynopsis);
$xml2 = getReplacedSynopsisXml($xml2);

return $xml1 === $xml2;
```

But it still doesn't disregard whitespace changes - I guess because `$originalSynopsis` keeps the whitespaces. As far as I see that's why we have to first convert the 2 `DOMElement`s to strings, and then load them to the comparator.

Reply:
Okay, let's stick with what you have then. Maybe rename the function to make it clear that it also modifies, something like `replaceAndCompareXml()`? Can't think of a great name...

Reply:
`replaceAndCompareXml()` is not that bad of a name - I don't have any better suggestion.